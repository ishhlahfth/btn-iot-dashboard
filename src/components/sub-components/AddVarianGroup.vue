<template>
  <help-modal v-model="modal.sm" permanent>
    <confirmation
      title="Delete Option Varian"
      message="Are you sure you want to delete this option varian ?"
      @close="modal.sm = false"
      @cancel="modal.sm = false"
      @confirm="deleteOption"
    />
  </help-modal>
  <help-modal v-model="modal.varian" permanent>
    <confirmation
      title="Delete Varian Group"
      :message="`Are you sure you want to delete ${data.name} ?`"
      @close="modal.varian = false"
      @cancel="modal.varian = false"
      @confirm="deleteVarianGroup"
    />
  </help-modal>
  <div class="inner-modal-fixed overflow-auto modal-sm px-1 pb-4">
    <div class="divide-y divide-grey-4 pb-3">
      <div class="flex justify-between pb-3 items-center">
        <icon
          @click="handleBack"
          name="chevron-left"
          class="text-blue-500 cursor-pointer"
          :size="10"
        />
        <span v-if="!isChanged" class="font-semibold text-heading4">Add Varian Group</span>
        <span v-if="isChanged" class="font-semibold text-heading4">Change Varian Group</span>
        <span>
          <icon
          v-if="isChanged"
          @click="modal.varian = true"
          name="trash-outline"
          class="text-red-500 cursor-pointer"
          :size="8"
          />
        </span>
      </div>
      <p></p>
    </div>
    <div class="grid gap-3 mb-3">
      <span class="font-semibold">Variation Group Name</span>
      <help-input class="w-full col-span-3" type="text" placeholder="Catalog name here" v-model="nameGroup" />
    </div>
    <section class="grid gap-3">
      <span class="font-semibold">Varian Options</span>
      <div v-for="(el, i) in options" :key="i" class="flex items-center">
        <help-input
          class="w-full col-span-3"
          type="text"
          placeholder="Catalog name here"
          v-model="el.name"
          :disabled="el.disable"
          :background="el.bgColor"
        >
          <template v-slot>
            <div class="flex items-center justify-end">
              <icon
                v-if="el.disable"
                name="minus-circle"
                :size="6"
                class="float-right cursor-pointer text-red-600 shadow-sm hover:text-red-400"
                @click="confirmDelete(el)"
              />
              <icon
                v-else
                name="plus-circle"
                :size="5"
                class="float-right cursor-pointer text-mint shadow-sm hover:text-mint-dark"
                @click="handleAddOption(el)"
              />
            </div>
          </template>
        </help-input>
      </div>
    </section>
    <div>
      <div
        v-if="!isChanged"
        @click="addVarianGorup"
        class="w-full flex justify-center items-center bg-blue-500 my-10 py-2 px-2 cursor-pointer rounded-2xl text-white hover:shadow-md"
      >
        <help-button bg-color="blue-500" icon="plus-circle" icon-only />
        <span class="text-center text-medium font-medium">Add Varian Group</span>
      </div>
      <div
        v-if="isChanged"
        @click="changeVarianGorup"
        class="w-full flex justify-center items-center bg-blue-500 my-10 py-2 px-2 cursor-pointer rounded-2xl text-white hover:shadow-md"
      >
        <help-button bg-color="blue-500" icon="plus-circle" icon-only />
        <span class="text-center text-medium font-medium">Add Varian Group</span>
      </div>
    </div>
  </div>
</template>

<script>
import { useToast } from 'vue-toastification';
import Icon from '@/components/atoms/Icon.vue';
import HelpInput from '@/components/atoms/Input.vue';
import HelpButton from '@/components/atoms/Button.vue';
import HelpModal from '@/components/templates/Modal.vue';
import Confirmation from '@/components/modals/Confirmation.vue';
import API from '../../apis';

export default {
  name: 'AddVarianGroup',
  emits: ['closeAddVarian', 'refetch'],
  components: {
    Icon,
    HelpInput,
    HelpButton,
    HelpModal,
    Confirmation,
  },
  props: {
    isChanged: {
      type: Boolean,
      default: false,
    },
    data: {
      type: Object,
      default: () => {},
    },
  },
  setup() {
    const toast = useToast();
    return { toast };
  },
  data() {
    return {
      nameGroup: '',
      payload: {},
      modal: {
        sm: false,
        varian: false,
      },
      options: [
        {
          name: '',
          description: '',
          disable: false,
          bgColor: 'white',
        },
      ],
    };
  },
  computed: {
    merchant() {
      return this.$store.state.merchant;
    },
    screenWidth() {
      return this.$store.state.screenWidth;
    },
  },
  methods: {
    async addVarianGorup() {
      const filter = this.options.filter((el) => el.name);
      const payload = {
        name: this.nameGroup,
        description: this.nameGroup,
        merchant_id: this.merchant.merchant_id,
        options: filter.map((el) => ({
          name: el.name,
          description: el.description,
        })),
      };
      try {
        const {
          data: { data },
        } = await API.post('variations', payload);
        this.toast.success(`Successfully add varian ${data.name} !`);
      } catch (error) {
        this.toast.error(error.response.data.meta.message);
      }
      this.$emit('closeAddVarian');
      this.$emit('refetch');
    },
    async changeVarianGorup() {
      const filter = this.options.filter((el) => el.name);
      const payload = {
        name: this.nameGroup,
        description: this.nameGroup,
        merchant_id: this.merchant.merchant_id,
        options: filter.map((el) => ({
          ...el,
          id: el.id ? el.id : null,
          name: el.name,
          description: el.description,
        })),
      };
      try {
        const {
          data: { data },
        } = await API.patch(`variations/${this.data.id}`, payload);
        this.toast.success(`Successfully update varian ${data.name} !`);
      } catch (error) {
        this.toast.error(error.response.data.meta.message);
      }
      this.$emit('closeAddVarian');
      this.$emit('refetch');
    },
    handleAddOption(payload) {
      payload.description = payload.name;
      payload.bgColor = 'gray-100';
      payload.disable = true;
      this.options.unshift({
        name: '',
        description: '',
      });
    },
    async deleteVarianGroup() {
      try {
        await API.delete(`variations/${this.data.id}`);
        this.toast.success(`Successfully delete option ${this.data.name}`);
        this.modal.varian = false;
        this.handleBack();
      } catch (error) {
        this.toast.error(error.response.data.meta.message);
      }
    },
    async deleteOption() {
      if (this.payload.id) {
        try {
          await API.delete(`/variation-options/${this.payload.id}`);
          this.toast.success(`Successfully delete option ${this.payload.name}`);
        } catch (error) {
          this.toast.error(error.response.data.meta.message);
        }
        this.options = this.options.filter((el) => el.name !== this.payload.name);
      } else {
        this.options = this.options.filter((el) => el.name !== this.payload.name);
      }
      this.modal.sm = false;
    },
    confirmDelete(payload) {
      this.modal.sm = true;
      this.payload = payload;
    },
    handleBack() {
      this.$emit('closeAddVarian');
      this.$emit('refetch');
    },
  },
  mounted() {
    if (this.isChanged) {
      this.nameGroup = this.data.name;
      this.options = this.data.options.map((el) => ({
        ...el,
        disable: true,
        bgColor: 'gray-100',
      }));
      this.options.unshift({
        name: '',
        description: '',
        disable: false,
        bgColor: 'white',
      });
    }
  },
};
</script>

<style></style>
