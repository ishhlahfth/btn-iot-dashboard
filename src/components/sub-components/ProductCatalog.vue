<template>
  <help-modal v-model="modal.sm" permanent>
    <confirmation
      title="Delete Item Catalog"
      :message="`Are you sure you want to delete item ${payloadCatalog.name} ?`"
      bg-color="red-500"
      @close="modal.sm = false"
      @cancel="modal.sm = false"
      @confirm="deleteCatalog"
      @load="loading.delete"
    />
  </help-modal>
  <div class="inner-modal-fixed modal-md overflow-auto px-1 flex flex-col sm:grid sm:gap-4">
    <div class="divide-y divide-grey-4">
      <div class="flex justify-between pb-3">
        <span class="font-semibold text-heading4">PRODUCT CATALOG</span>
      </div>
      <p></p>
    </div>
    <form @submit.prevent="addCatalog" class="grid grid-cols-4 gap-2 sm:py-0 py-3">
      <help-input
        v-model="newCatalog"
        class="w-full col-span-3"
        type="text"
        placeholder="Catalog name here"
      />
      <help-button
        bg-color="blue-500"
        icon="plus-circle"
        label="Add"
        type="submit"
        :loading="loading.add"
      />
    </form>
    <div class="grid gap-2 sm:gap-4 sm:py-0 py-2">
      <template v-if="itemCatalogs.length">
        <div v-for="(itemCatalog, i) in itemCatalogs" :key="i" class="flex items-center">
          <help-input
            class="w-full"
            type="text"
            v-model="itemCatalog.label"
            :background="itemCatalog.color"
            :disabled="itemCatalog.disabled"
          >
            <template v-slot>
              <div class="flex items-center justify-end">
                <div v-if="itemCatalog.editable">
                  <icon
                    name="plus-circle"
                    :size="5"
                    class="float-right cursor-pointer hover:text-blue-500"
                    @click="patchCatalog"
                  />
                </div>
                <div v-else class="flex items-center justify-end">
                  <icon
                    name="edit"
                    :size="5"
                    class="cursor-pointer hover:text-mint"
                    @click="handleEditCatalog(itemCatalog)"
                  />&nbsp;&nbsp;
                  <icon
                    name="trash"
                    :size="5"
                    class="cursor-pointer hover:text-red-600"
                    @click="confirmDelete(itemCatalog)"
                  />
                </div>
              </div>
            </template>
          </help-input>
        </div>
      </template>
      <template v-else>
        <p class="sm:px-2 sm:ml-2 py-2 text-gold-dark bg-gold-soft rounded-sm text-center inline">
          No variant group found please add some
        </p>
      </template>
    </div>
    <div class="divide-y divide-grey-4 sm:py-0 py-3">
      <p></p>
      <div class="flex justify-end pt-3">
        <help-button
          bg-color="bg-white"
          color="black-500"
          label="Close"
          class="border"
          @click="$emit('close')"
        />
      </div>
    </div>
  </div>
</template>

<script>
import { useToast } from 'vue-toastification';
import Icon from '@/components/atoms/Icon.vue';
import HelpInput from '@/components/atoms/Input.vue';
import HelpButton from '@/components/atoms/Button.vue';
import HelpModal from '@/components/templates/Modal.vue';
import Confirmation from '@/components/modals/Confirmation.vue';
import API from '../../apis';

export default {
  name: 'ProductCatalog',
  emits: ['close', 'handleAddCatalog'],
  components: {
    Icon,
    HelpInput,
    HelpButton,
    HelpModal,
    Confirmation,
  },
  props: {
    itemCatalogs: {
      type: Array,
      default: () => [],
    },
  },
  setup() {
    const toast = useToast();
    return { toast };
  },
  data() {
    return {
      newCatalog: '',
      modal: {
        sm: false,
      },
      payloadCatalog: {},
      loading: {
        delete: false,
        add: false,
      },
    };
  },
  computed: {
    merchant() {
      return this.$store.state.merchant;
    },
    screenWidth() {
      return this.$store.state.screenWidth;
    },
  },
  methods: {
    async addCatalog() {
      this.loading.add = true;
      const payload = {
        name: this.newCatalog,
        description: null,
        merchant_id: this.merchant.merchant_id,
        status: 'AVAILABLE',
        sort_no: 1,
      };
      try {
        const {
          data: { data },
        } = await API.post('catalogs', payload);
        this.toast.success(`Successfully add item ${data.name} in catalogs !`);
        this.newCatalog = '';
        this.$emit('handleAddCatalog');
      } catch (error) {
        this.toast.error(error.response.data.meta.message);
      }
      this.loading.add = false;
    },
    async deleteCatalog() {
      this.loading.delete = true;
      try {
        await API.delete(`catalogs/${this.payloadCatalog.id}`);
        this.toast.success(`Successfully delete item ${this.payloadCatalog.name} in catalogs !`);
        this.$emit('handleAddCatalog');
        this.modal.sm = false;
      } catch (error) {
        this.toast.error(error.response.data.meta.message);
      }
      this.loading.delete = false;
    },
    async patchCatalog() {
      const payload = {
        name: this.payloadCatalog.label,
        description: null,
        merchant_id: this.merchant.merchant_id,
        status: 'AVAILABLE',
        sort_no: 1,
      };
      try {
        const {
          data: { data },
        } = await API.patch(`catalogs/${this.payloadCatalog.id}`, payload);
        this.toast.success(`Successfully edit item ${data.name} in catalogs !`);
        this.$emit('handleAddCatalog');
      } catch (error) {
        this.toast.error(error.response.data.meta.message);
      }
    },
    handleEditCatalog(payload, param) {
      this.payloadCatalog = payload;
      if (param === 'edit-name') {
        payload.color = 'gray-100';
        payload.editable = false;
        payload.disabled = true;
        this.patchCatalog();
      } else {
        payload.color = 'white';
        payload.editable = true;
        payload.disabled = false;
      }
    },
    confirmDelete(payload) {
      this.modal.sm = true;
      this.payloadCatalog = payload;
    },
  },
};
</script>

<style></style>
