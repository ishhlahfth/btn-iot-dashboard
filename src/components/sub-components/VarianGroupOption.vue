<template>
  <add-varian-group v-if="flagAddVarian" @closeAddVarian="flagAddVarian = false" />
  <div v-else class="inner-modal-fixed overflow-auto px-1">
    <div class="divide-y divide-grey-4 pb-3">
      <div class="flex justify-between pb-3 items-center">
        <icon
          @click="$emit('closeSelectVarian')"
          name="chevron-left"
          class="text-blue-500 cursor-pointer"
          :size="10"
        />
        <span class="font-semibold text-heading4">Select Varian Group</span>
        <span></span>
      </div>
      <p></p>
    </div>
    <div class="grid gap-3 my-2">
      <div class="py-3 px-3 w-full rounded-md cursor-pointer" style="background: #F6F6F6">
        <div class="flex justify-between items-center">
          <span class="font-semibold">Ukuran</span>
          <span class="text-blue-500 font-medium">Change</span>
        </div>
        <div class="grid grid-cols-4 gap-1 py-2">
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Small</span
          >
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Large</span
          >
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Extra Large</span
          >
        </div>
      </div>
      <div class="py-3 px-3 w-full rounded-md cursor-pointer" style="background: #F6F6F6">
        <div class="flex justify-between items-center">
          <span class="font-semibold">Topping Makanan</span>
          <span class="text-blue-500 font-medium">Change</span>
        </div>
        <div class="grid grid-cols-4 gap-1 py-2">
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Small</span
          >
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Large</span
          >
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Extra Large</span
          >
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Extra Large</span
          >
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Extra Large</span
          >
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Extra Large</span
          >
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Extra Large</span
          >
        </div>
      </div>
      <div class="py-3 px-3 w-full rounded-md cursor-pointer" style="background: #F6F6F6">
        <div class="flex justify-between items-center">
          <span class="font-semibold">Ukuran</span>
          <span class="text-blue-500 font-medium">Change</span>
        </div>
        <div class="grid grid-cols-4 gap-1 py-2">
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Small</span
          >
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Large</span
          >
          <span
            class="text-center text-small font-medium bg-blue-500 py-2 px-2 rounded-2xl text-white hover:shadow-md"
            >Extra Large</span
          >
        </div>
      </div>
    </div>
    <div
      @click="flagAddVarian = true"
      class="w-full flex justify-center items-center bg-blue-500 my-10 py-2 px-2 cursor-pointer rounded-2xl text-white hover:shadow-md"
    >
      <help-button bg-color="blue-500" icon="plus-circle" icon-only />
      <span class="text-center text-small font-medium">Add Variation Group</span>
    </div>
    <div>
      <div
        @click="$emit('closeSelectVarian')"
        class="w-full flex justify-center items-center bg-blue-500 my-10 py-2 px-2 cursor-pointer rounded-2xl text-white hover:shadow-md"
      >
        <span class="text-center text-medium font-medium">Save</span>
      </div>
    </div>
  </div>
</template>

<script>
import Icon from '@/components/atoms/Icon.vue';
import HelpButton from '@/components/atoms/Button.vue';
import AddVarianGroup from '@/components/sub-components/AddVarianGroup.vue';

export default {
  name: 'VarianGroupOption',
  components: {
    Icon,
    HelpButton,
    AddVarianGroup,
  },
  data() {
    return {
      flagAddVarian: false,
    };
  },
};
</script>

<style></style>
