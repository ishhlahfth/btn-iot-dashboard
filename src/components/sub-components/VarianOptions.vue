<template>
  <help-modal v-model="modal.sm" permanent>
    <confirmation
      title="Delete Varian Group"
      :message="`Are you sure you want to delete this ${payloadVarian?.name} group ?`"
      @close="modal.sm = false"
      @cancel="modal.sm = false"
      @confirm="handleDeleteVarian"
    />
  </help-modal>
  <help-modal v-model="modal.close" permanent>
    <confirmation
      title="Cancel Add Varian Group"
      :message="`Are you sure you want to cancel add varian group ?`"
      @close="modal.close = false"
      @cancel="modal.close = false"
      @confirm="$emit('closeVarian')"
    />
  </help-modal>
  <varian-group-option
    v-if="flagSelectVarianGroup"
    @closeSelectVarian="flagSelectVarianGroup = false"
    @selectVarian="handleSelectVarian"
    :data="payloadVarianGroup"
    :isEdit="isEdit"
  />
  <div v-else class="inner-modal-fixed overflow-auto modal-md px-1">
    <div class="divide-y divide-grey-4 pb-3">
      <div class="flex justify-between pb-3">
        <span class="font-semibold text-heading4">Varian Options</span>
      </div>
      <p></p>
    </div>
    <section class="grid gap-3 py-2">
      <div
        v-for="(itemVarian, i) in selectVarians"
        :key="i"
        class="border rounded-md p-4 grid gap-3 shadow-md"
      >
        <div class="flex justify-between">
          <div class="w-full">
            <span v-if="itemVarian" class="font-semibold">Varian {{ i + 1 }}</span>
            <div v-else class="rounded bg-grey-4 h-6 w-32 animate-pulse"></div>
          </div>
          <span
            @click="confirmDelete(itemVarian)"
            v-if="itemVarian?.name"
            class="font-semibold text-red-600 cursor-pointer"
            >Delete</span
          >
          <span v-if="!itemVarian?.name" class="font-semibold text-grey-4 cursor-pointer"
            >Delete</span
          >
        </div>
        <div
          class="py-2 px-2 w-full rounded-md flex justify-between items-center cursor-pointer"
          style="background: #F6F6F6"
          @click="handleAddVarian(false, itemVarian)"
        >
          <div class="grid grid-flow-row gap-1">
            <span class="font-semibold">Grup Variasi</span>
            <span v-if="!itemVarian?.name" class="text-gray-500">Belum Pilih Grup Variasi</span>
            <span v-if="itemVarian?.name" class="text-gray-700">{{ itemVarian?.name }}</span>
          </div>
          <icon name="chevron-right" class="text-blue-500" :size="10" />
        </div>
        <template v-if="itemVarian?.name">
          <div class="flex justify-between">
            <span class="text-gray-500 text-sm">Is varian required ?</span>
            <help-toggle
              v-model="itemVarian.is_mandatory"
              @change="itemVarian.is_mandatory = !itemVarian.is_mandatory"
            />
          </div>
        </template>
        <template v-if="itemVarian?.name">
          <div class="flex justify-between">
            <span class="text-gray-500">Choosed more than 1 choice ?</span>
            <help-toggle
              v-model="itemVarian.multiple_choice"
              @change="itemVarian.multiple_choice = !itemVarian.multiple_choice"
            />
          </div>
        </template>
        <div v-if="itemVarian?.options" class="grid grid-flow-col divide-y divide-grey-4">
          <p></p>
          <div class="grid gap-2 py-3">
            <div class="flex pb-2">
              <span class="font-semibold">Varian Options</span>
            </div>
            <div v-for="item in itemVarian.options" :key="item.id" class="grid grid-flow-col gap-3">
              <help-checkbox v-model:checked="item.picked" />
              <help-input
                type="text"
                placeholder="Varian item name here"
                v-model="item.name"
                disabled
              />
              <help-input v-model="item.price" type="number" placeholder="Additional price here" />
            </div>
          </div>
        </div>
      </div>
    </section>
    <section class="my-5">
      <span
        v-if="selectVarians[0].name"
        @click="handleAddVarian(true)"
        class="font-medium bg-blue-500 shadow-sm py-2 px-6 rounded-2xl text-white cursor-pointer border hover:shadow-md flex items-center justify-center"
        ><icon name="plus-circle" :size="6" />&nbsp;Add Varian</span
      >
      <span
        v-if="!selectVarians[0].name"
        class="font-medium bg-grey-4 shadow-sm py-2 px-6 rounded-2xl text-white cursor-pointer border hover:shadow-md flex items-center justify-center"
        ><icon name="plus-circle" :size="6" />&nbsp;Add Varian</span
      >
    </section>
    <div class="divide-y divide-grey-4 py-8">
      <p></p>
      <div class="flex justify-end pt-8">
        <span
          @click="modal.close = true"
          class="font-medium shadow-sm py-2 px-6 rounded-2xl cursor-pointer border hover:shadow-md"
          >Cancel</span
        >&nbsp;
        <span
          v-if="selectVarians[0].name"
          @click="handleSaveVarian"
          class="font-medium bg-blue-500 shadow-sm py-2 px-6 rounded-2xl text-white cursor-pointer border hover:shadow-md"
          >Save Product</span
        >
        <span
          v-if="!selectVarians[0].name"
          class="font-medium bg-grey-4 shadow-sm py-2 px-6 rounded-2xl text-white cursor-pointer border hover:shadow-md"
          >Save Product</span
        >
      </div>
    </div>
  </div>
</template>

<script>
import Icon from '@/components/atoms/Icon.vue';
import HelpToggle from '@/components/atoms/Toggle.vue';
import HelpInput from '@/components/atoms/Input.vue';
import HelpCheckbox from '@/components/atoms/Checkbox.vue';
import VarianGroupOption from '@/components/sub-components/VarianGroupOption.vue';
import HelpModal from '@/components/templates/Modal.vue';
import Confirmation from '@/components/modals/Confirmation.vue';

export default {
  name: 'VarianOptions',
  emits: ['closeVarian', 'getSelectVarian'],
  components: {
    HelpToggle,
    HelpInput,
    HelpCheckbox,
    VarianGroupOption,
    Icon,
    HelpModal,
    Confirmation,
  },
  props: {
    flagEditVarian: {
      type: Boolean,
      default: false,
    },
    data: {
      type: Array,
      default: () => [],
    },
    isEdit: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      isTrue: true,
      flagSelectVarianGroup: '',
      form: {
        src: '',
        productCatalog: 'Pilih Katalog',
        catalog_id: 'Pilih Kategori',
        status: 'Pilih Status',
        varianGroup: {
          name: '',
          required: false,
          multipleChoice: false,
        },
      },
      selectVarians: [
        {
          name: '',
        },
      ],
      modal: {
        sm: false,
        close: false,
      },
      isAdded: false,
      indexAdded: 0,
      payloadVarian: {},
      payloadVarianGroup: [],
      tempFlag: {},
    };
  },
  computed: {
    merchant() {
      return this.$store.state.merchant;
    },
    screenWidth() {
      return this.$store.state.screenWidth;
    },
  },
  mounted() {
    if (this.data.length) {
      this.selectVarians = this.data.map((el) => ({
        ...el,
        options: el.options.map((e) => ({
          ...e,
          picked: true,
        })),
      }));
      this.payloadVarianGroup = this.selectVarians;
    }
  },
  methods: {
    handleSelectVarian(payload) {
      if (this.isAdded) {
        this.indexAdded += 1;
        this.selectVarians.push(payload);
      } else {
        this.selectVarians[this.indexAdded] = payload;
      }
      console.log(payload, 'select varian');
      if (this.payloadVarianGroup.length === 0) {
        this.payloadVarianGroup.push(payload);
      } else {
        switch (this.isAdded) {
          case true:
            this.payloadVarianGroup.push(payload);
            break;
          default:
            this.payloadVarianGroup[this.payloadVarianGroup.findIndex((el) => el.id === this.tempFlag.id)] = payload;
            break;
        }
      }
    },
    handleAddVarian(param, payload) {
      this.flagSelectVarianGroup = true;
      this.isAdded = param;
      if (payload) {
        if (payload.name) {
          this.tempFlag = payload;
        }
      }
    },
    handleDeleteVarian() {
      if (this.selectVarians.length === 1) {
        this.selectVarians = [
          {
            name: '',
          },
        ];
        this.payloadVarianGroup = [];
        this.indexAdded = 0;
      } else {
        this.selectVarians = this.selectVarians.filter((el) => el.id !== this.payloadVarian.id);
        this.payloadVarianGroup = this.payloadVarianGroup.filter((el) => el.id !== this.payloadVarian.id);
      }
      this.modal.sm = false;
    },
    confirmDelete(payload) {
      this.payloadVarian = payload;
      this.modal.sm = true;
    },
    handleSaveVarian() {
      const payloadToSend = this.selectVarians.map((el) => ({
        ...el,
        is_mandatory: el.is_mandatory || false,
        multiple_choice: el.multiple_choice || false,
        sort_no: 1,
        options: el.options.map((e) => ({
          ...e,
          price: +e.price || 0,
          is_mandatory: false,
          is_default: false,
          picked: e.picked || false,
        })),
      }));
      this.$emit('getSelectVarian', payloadToSend);
      this.$emit('closeVarian');
    },
  },
};
</script>

<style></style>
