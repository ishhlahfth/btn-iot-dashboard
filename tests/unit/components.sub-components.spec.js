import AddVarianGroup from '@/components/sub-components/AddVarianGroup.vue';
import ProductCatalog from '@/components/sub-components/ProductCatalog.vue';
import VarianGroupOption from '@/components/sub-components/VarianGroupOption.vue';
import VarianOptions from '@/components/sub-components/VarianOptions.vue';

describe('AddVarianGroup.vue', () => {
  it('should render correctly', () => {
    expect(AddVarianGroup.$el).toMatchSnapshot();
  });

  it('should render name AddVarianGroup', () => {
    expect(AddVarianGroup.name).toBe('AddVarianGroup');
  });
});

describe('ProductCatalog.vue', () => {
  it('should render correctly', () => {
    expect(ProductCatalog.$el).toMatchSnapshot();
  });

  it('should render name ProductCatalog', () => {
    expect(ProductCatalog.name).toBe('ProductCatalog');
  });
});

describe('VarianGroupOption.vue', () => {
  it('should render correctly', () => {
    expect(VarianGroupOption.$el).toMatchSnapshot();
  });

  it('should render name VarianGroupOption', () => {
    expect(VarianGroupOption.name).toBe('VarianGroupOption');
  });
});

describe('VarianOptions.vue', () => {
  it('should render correctly', () => {
    expect(VarianOptions.$el).toMatchSnapshot();
  });

  it('should render name VarianOptions', () => {
    expect(VarianOptions.name).toBe('VarianOptions');
  });
});
