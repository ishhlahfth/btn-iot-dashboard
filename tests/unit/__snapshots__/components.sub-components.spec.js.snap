// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddVarianGroup.vue should render correctly 1`] = `undefined`;

exports[`ProductCatalog.vue should render correctly 1`] = `undefined`;

exports[`VarianGroupOption.vue should render correctly 1`] = `undefined`;

exports[`VarianOptions.vue should render correctly 1`] = `undefined`;
